codegen-server/src/main/kotlin/software/amazon/smithy/rust/codegen/server/smithy/generators/ServerOperationHandlerGenerator.kt

```diff
@@ -0,0 +1,222 @@
+
+import software.amazon.smithy.model.shapes.OperationShape
+import software.amazon.smithy.rust.codegen.rustlang.CargoDependency
+import software.amazon.smithy.rust.codegen.rustlang.RustWriter
+import software.amazon.smithy.rust.codegen.rustlang.asType
+import software.amazon.smithy.rust.codegen.rustlang.rust
+import software.amazon.smithy.rust.codegen.rustlang.rustBlockTemplate
+import software.amazon.smithy.rust.codegen.rustlang.rustTemplate
+import software.amazon.smithy.rust.codegen.server.smithy.ServerCargoDependency
+import software.amazon.smithy.rust.codegen.server.smithy.ServerRuntimeType
+import software.amazon.smithy.rust.codegen.server.smithy.protocols.ServerHttpProtocolGenerator
+import software.amazon.smithy.rust.codegen.smithy.CodegenContext
+import software.amazon.smithy.rust.codegen.smithy.RuntimeType
+import software.amazon.smithy.rust.codegen.smithy.generators.error.errorSymbol
+import software.amazon.smithy.rust.codegen.util.outputShape
+
+/**
+ * ServerOperationHandlerGenerator
+ */
+class ServerOperationHandlerGenerator(
+    codegenContext: CodegenContext,
+    private val operations: List<OperationShape>,
+) {
+    private val serverCrate = "aws_smithy_http_server"
+    private val service = codegenContext.serviceShape
+    private val model = codegenContext.model
+    private val symbolProvider = codegenContext.symbolProvider
+    private val operationNames = operations.map { symbolProvider.toSymbol(it).name }
+    private val runtimeConfig = codegenContext.runtimeConfig
+    private val codegenScope = arrayOf(
+        "Axum" to ServerCargoDependency.Axum.asType(),
+        "PinProject" to ServerCargoDependency.PinProject.asType(),
+        "Tower" to ServerCargoDependency.Tower.asType(),
+        "FuturesUtil" to ServerCargoDependency.FuturesUtil.asType(),
+        "SmithyHttpServer" to CargoDependency.SmithyHttpServer(runtimeConfig).asType(),
+        "SmithyRejection" to ServerHttpProtocolGenerator.smithyRejection(runtimeConfig),
+        "Phantom" to ServerRuntimeType.Phantom,
+        "http" to RuntimeType.http,
+    )
+
+    fun render(writer: RustWriter) {
+        renderStaticRust(writer)
+        renderHandlersImpl(writer)
+    }
+
+    /*
+     * Renders the implementation of the Handler trait for all available operations.
+     * Handlers are implemented for FnOnce signatures with and without state management.
+     */
+    private fun renderHandlersImpl(writer: RustWriter) {
+        operations.map { operation ->
+            val operationName = symbolProvider.toSymbol(operation).name
+            val inputName = "crate::input::${operationName}Input"
+            val inputWrapperName = "crate::operation::$operationName${ServerHttpProtocolGenerator.OPERATION_INPUT_WRAPPER_SUFFIX}"
+            val outputWrapperName = "crate::operation::$operationName${ServerHttpProtocolGenerator.OPERATION_OUTPUT_WRAPPER_SUFFIX}"
+            val stateList = listOf<Boolean>(true, false)
```

Comment:
Maybe it's cleaner if this function takes in the boolean and the caller calls it twice.

Reply:
You are probably right, I'll refactor.

Reply:
Done